[Derrick Wong Liang Jun] iP

LemonDrew opened the pull request:
# Derrick's Chatbot
## This chatbot allows you to keep track of the list of items. The chatbot is :
* Easy to learn 🔢 
* Text-based 🫨 
* Extremely fast to use!!!! 😆 

> [!IMPORTANT]
Here are the list of **commands** that the chatbot currently supports:
1. list
2. mark
3. unmark
4. delete
5. find
6. todo
7. deadline
8. event

## Here is an example of the Task List that the chatbot will be using.
- [x] borrow books
- [ ] Meeting (/from 6am /to 7am)
- [ ] return books (/by 12pm)


## In order to run the chatbot, please navigate to the directory containing the chatbot jar file and run the command:
`java -jar "{filename}.jar"`

## Here is a snippet of the code used to run the chatbot:
```java
public static void main(String[] args) {
        Derrick chatbot = new Derrick();
        chatbot.run();
    }
```

For more details, please click [here](https://github.com)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LemonDrew committed:
Add greetings and exit

LemonDrew committed:
Add echoing of user input

LemonDrew committed:
Add Todo List and displaying of the list

LemonDrew committed:
Add ability to label tasks as done/undone

LemonDrew committed:
Add 3 types of items (todo, deadline, event)

LemonDrew committed:
Add exception handling (for invalid input)

LemonDrew committed:
Add deletion of items from the list

LemonDrew committed:
Add test cases to test chatbot

LemonDrew committed:
Improve exception handling by adding exception classes to handle specific errors

LemonDrew committed:
Add Exception class for handling deletion from an empty list

LemonDrew committed:
Add Exception class for handling missing description for items

LemonDrew committed:
Add Exception class for handling selection of invalid items in the list

LemonDrew committed:
Add Exception class for handling missing position in both mark and unmark prompts

LemonDrew committed:
Add ENUMS class to better manage the commands needed to execute the chatbot (mark, unmark, delete, list, todo, event, deadline)

LemonDrew committed:
Add changeFormat() methods to Task subclasses

LemonDrew committed:
Fix bugs regarding changing isDone status

LemonDrew committed:
Add data storage feature for storing list

LemonDrew committed:
Merge branch 'branch-Level-7'

LemonDrew committed:
Convert "yyyy-mm-dd" to "MMM dd yyyy" format

LemonDrew committed:
Merge branch 'branch-Level-8'

LemonDrew committed:
Add and improve OOP

LemonDrew committed:
Merge branch 'branch-A-MoreOOP'

LemonDrew committed:
Add packages

LemonDrew committed:
Merge branch 'branch-A-Packages'

LemonDrew committed:
Merge remote-tracking branch 'origin/add-gradle-support'

LemonDrew committed:
Set up JUnit for testing

LemonDrew committed:
Add Junit Tests for TaskList

LemonDrew committed:
Add JUnit Tests for "Commands" Enums class

LemonDrew committed:
Merge branch 'branch-A-JUnit'

LemonDrew committed:
Create Jar File for Chatbot

LemonDrew committed:
Add find function

LemonDrew committed:
Merge branch 'branch-Level-9'

LemonDrew committed:
Fix coding style

LemonDrew committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/CommandHandler.java

LemonDrew committed:
Add JavaDocs

LemonDrew committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Commands.java
#	src/main/java/Task.java

LemonDrew committed:
Fix merge conflicts

LemonDrew committed:
Add GUI for chatbot

sarahteonin commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,19 @@
+     * @param command The String input entered by the user, which represents a command.
+     * @return A corresponding enum value indicating the command.
+     */
+    public static Commands fromString(String command) {
```
Could this command name be clear, such as fromStringToEnum?

sarahteonin commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     * @return ArrayList<Task> result.
+     * @throws RuntimeException If an I/O error occurs during the retrieval of the directory or file.
+     */
+    public ArrayList<Task> loadTasksFromFile() {
```
I like your clear naming of your method here!

sarahteonin commented on src/main/java/Derrick.java:
```diff
@@ -0,0 +1,28 @@
+
+
+public class Derrick {
+    private ArrayList<Task> toDo = new ArrayList<>();
```
Should this be named something else to be clearer, for example taskList?

sarahteonin commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,41 @@
+
+    String start;
+    String end;
+    LocalDate newStart;
```
Could newStart give a different impression of what this variable represents, would formattedStart be better?

sarahteonin reviewed (commented):
good job on your code! just a few minor mistakes 👍

LemonDrew committed:
Update images

LemonDrew committed:
Merge branch 'branch-Level-10'

LemonDrew committed:
Add checkStyle to detect coding style violations

LemonDrew committed:
Merge branch 'branch-A-CheckStyle'

LemonDrew committed:
Fix coding violations : use CodeStyle to detect violations

Many coding style violations in the current code

Fixing these coding style violations allows for standardization and
better readability

Let's use the CodeStyle plugins to detect such coding violations
and amend them accordingly

LemonDrew committed:
Merge pull request #1 from LemonDrew/branch-A-CodeQuality

Fix coding violations : use CodeStyle to detect violations

LemonDrew committed:
Derrick class: add assertions to document important assumptions

There are many assumptions made during the initialization of the
Derrick class.

The assumptions if left unchecked could cause problems in the
future

Adding assertions allow us to ensure that these assumptions are
documented well. For instance, adding assertions to ensure that
both the tasks and storage are initialized correctly.

Let's add assertions to check whether the tasks and storage fields
are initialized correctly

LemonDrew committed:
Merge pull request #2 from LemonDrew/branch-A-Assertions

Derrick class: add assertions to document important assumptions

LemonDrew committed:
Handle duplicate tasks in CommandHandler class

There are cases where the user adds duplicate tasks, making it
difficult to distinguish between the duplicates.

Identifying these cases allow us to inform the user that
duplicate items are being added

Let's add a DuplicateItemsException class to handle cases where
duplicate tasks are being added to the TaskList

LemonDrew committed:
Merge branch 'Branch-C-Duplicates'

BlazeChron commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class Deadline extends Task {
+
+    private LocalDate newBy;
```
Could this have a more appropriate descriptive name?

BlazeChron commented on src/main/java/DuplicateItemException.java:
```diff
@@ -0,0 +1,6 @@
+package ip.derrick;
+public class DuplicateItemException extends Exception{
```
Perhaps add a javadoc comment here as well?

BlazeChron commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,51 @@
+
+    private String start;
+    private String end;
+    private LocalDate newStart;
+    private LocalDate newEnd;
```
Could this have a more appropriate descriptive name?

BlazeChron commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,114 @@
+                System.out.println(e.getMessage());
+            }
+            break;
+        case UNMARK: {
+            try {
+                tasks.unmarkItem(input);
+                storage.saveTasksToFile(tasks);
+            } catch (MissingPositionException | MissingItemException e) {
+                System.out.println(e.getMessage());
+            }
+            break;
+        }
```
Perhaps the switch case statements could follow a consistent style?

BlazeChron reviewed (commented):
The code is great. A few updates to make the code more informative will help.

LemonDrew committed:
Improve code quality using AI

There might be hidden bugs or suboptimizations not detected

Let's use AI to scan through the code and look for potential bugs

LemonDrew committed:
Refactor code to improve readability

The current loadTaskFromFile() method is too long, which can lead
to potential bugs

Let's refactor the code by creating a helper method to better
categorize the tasks found in the data retrieved

LemonDrew committed:
Fix minor bugs in displaying of event

LemonDrew committed:
Merge branch 'branch-AiAssisted'

LemonDrew committed:
Add a User Guide for the chatbot

LemonDrew committed:
Update Ui.png for better readability and clarity

LemonDrew committed:
Update README.md to include more features of the chatbot

LemonDrew committed:
Add more JUnit tests to test the mark() method of TaskList class

LemonDrew committed:
Improve JavaDoc comments to follow the specified coding standard

LemonDrew committed:
Add greetings of user when the GUI is initialized

LemonDrew committed:
Fix minor bug in displaying of Event's description

LemonDrew committed:
Update README to include more features of the chatbot

LemonDrew committed:
Fix spelling errors in the README.md

LemonDrew committed:
Update README.md to including exit feature

LemonDrew committed:
Update README.md to add exit command

LemonDrew committed:
Add example of exit command in README

LemonDrew committed:
Add explanation of exit command in README.md

LemonDrew committed:
Fix Javadoc comment parsing error for Task return type

LemonDrew committed:
Add missing newline at the end of the file

LemonDrew committed:
Update archiveBaseName to Derrick

LemonDrew committed:
Move files to their designated package folders

LemonDrew committed:
Remove unnecessary import of package

LemonDrew committed:
Remove method that has no usage

LemonDrew committed:
Fix minor bug in unmarking item

LemonDrew committed:
Update README.md

LemonDrew committed:
Update README.md

LemonDrew committed:
Update README.md
